proxy/nasshp/nassh.go

```diff
@@ -326,18 +326,26 @@ func (np *NasshProxy) allow(r *http.Request, w http.ResponseWriter, sid, hostpor
 		logid = LogId(sid, r, hostport, creds)
 	}
 	if np.filter != nil {
-		verdict := np.filter("tcp", hostport, creds)
-		if HostIp(hostport).IsUrl() {
-			for _, u := range HostIp(hostport).Resolve() {
+		host, port, err := net.SplitHostPort(hostport)
+		if err != nil {
+			goto DENY
+		}
+		{
+			res, err := net.LookupHost(host)
+			if err != nil {
+				goto DENY
+			}
+			for _, u := range res {
 				// TODO(adam): make verdict merging configurable from ACL list
-				verdict = verdict.MergePreferAllow(np.filter("tcp", u, creds))
+				if verdict := np.filter("tcp", net.JoinHostPort(u, port), creds); verdict == VerdictAllow {
```

Comment:
problem: when the proxy connects below, DNS will pick one of the hosts at random. So... our ACLs should really check that all are allowed. Otherwise, one may create a DNS record that resolves to 4 names, 3 of which are ok, 1 of which is evil, and have 75% probability of getting through the ACLs, and 25% past that to actually be able to connect to the evil host, so... if he makes ~6 attempts, he'll have a reasonable probability one of them will succeed?

Reply:
Changed the ACL to only accept if all are allowed